src/router_core/router_core.c

```diff
@@ -896,6 +914,27 @@ void qdr_del_subscription_ref_CT(qdr_subscription_ref_list_t *list, qdr_subscrip
 }
 
 
+static void qdr_general_handler_cleanup(void *context)
+{
+    qdr_core_t              *core = (qdr_core_t*) context;
+    qdr_general_work_list_t  work_list;
+    qdr_general_work_t      *work;
+
+    sys_mutex_lock(core->work_lock);
+    DEQ_MOVE(core->work_list, work_list);
+    sys_mutex_unlock(core->work_lock);
+
+    work = DEQ_HEAD(work_list);
+    while (work) {
+        DEQ_REMOVE_HEAD(work_list);
+        work->handler(core, work);
+        qdr_do_message_to_addr_free(core, work);
```

Comment:
Isn't it possible that this will double free if work->handler is qdr_do_message_to_addr_free()?